waiter/src/waiter/process_request.clj

```diff
@@ -562,13 +583,21 @@
                               queue-timeout-ms (:queue-timeout-ms instance-request-properties)
                               instance (<? (prepare-instance instance-rpc-chan service-id reason-map start-new-service-fn request-state-chan
                                                              unable-to-create-instance queue-timeout-ms reservation-status-promise metric-group
-                                                             add-debug-header-into-response!))]
+                                                             add-debug-header-into-response!))
+                              request (-> request
+                                          (utils/mark-request-time :instance-reserved)
+                                          (assoc :instance-id (subs (:id instance) (inc (count service-id)))
+                                                 :instance-host (:host instance)
+                                                 :instance-port (:port instance)
+                                                 :instance-proto (:protocol instance)))]
```

Comment:
Can we destructure these guys?

Reply:
We can.  It feels awkward to me, but `id` `name` `port` and `protocol` are all available.

Reply:
Why awkward?

Reply:
The instance fields are now in scope for a larger function.  I did it anyway, and it feels less awkward.